sklearn/metrics/scorer.py

```diff
@@ -66,6 +67,34 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """Check if estimator has a classes argument and if scorer has a
+        labels argument and return new kwargs for scorer if label not already
+        set in make score initialization"""
+        est_has_classes = hasattr(estimator, "classes_")
+        scorer_has_labels = "labels" in signature(
+                                            self._score_func).parameters
+
+        if est_has_classes and scorer_has_labels:
+            classes = getattr(estimator, "classes_", None)
```

Comment:
Right, forgot that we are definitely after the fit that we want to evaluate. Not enough coffee.

And @jnothman indeed, this can be a separate enhancement. This PR was trying do draft the API that's required to make Everything Work™. It looks like we can port this bug-fix over without relying on ``classes`` in ``__init__``. But I'd rather get this PR somewhat complete (for a single estimator) before factoring out anything.